src/Surface.zig

```diff
@@ -2127,13 +2134,27 @@ fn resize(self: *Surface, size: rendererpkg.ScreenSize) !void {
     self.io.queueMessage(.{ .resize = self.size }, .unlocked);
 }
 
-/// Recalculate the balanced padding if needed.
-fn balancePaddingIfNeeded(self: *Surface) void {
-    if (!self.config.window_padding_balance) return;
+/// Recalculate and balance the padding if needed.
+fn updatePadding(self: *Surface, scrollbar_width: ?u32) void {
+    const new_scrollbar_width = new: {
+        if (scrollbar_width) |width| {
+            self.scrollbar_width = width;
+            break :new true;
+        } else {
+            break :new false;
+        }
+    };
+    if (!new_scrollbar_width and !self.config.window_padding_balance) return;
```

Comment:
I think you could check `window_padding_balance` in the else above and then avoid needing the `new_scrollbar_width` flag entirely.